third_party/move/move-vm/runtime/src/interpreter.rs

```diff
@@ -153,11 +160,20 @@ impl Interpreter {
             .enter_function(&current_frame, current_frame.function.as_ref())
             .map_err(|e| self.set_location(e))?;
         loop {
-            let resolver = current_frame.resolver(loader, module_store);
-            let exit_code =
-                current_frame //self
-                    .execute_code(&resolver, &mut self, data_store, module_store, gas_meter)
-                    .map_err(|err| self.attach_state_if_invariant_violation(err, &current_frame))?;
+            let resolver = current_frame
+                .resolver(loader, module_store, module_storage, script_storage)
+                .map_err(|e| e.finish(Location::Undefined))?;
+            let exit_code = current_frame //self
+                .execute_code(
```

Comment:
why not extract module_storage (and even before, module_store) out of the resolver and not have to pass to .execute_code?

Reply:
I think mostly because I have to use dyn over impl everywhere but I guess that should be better.

Reply:
yeah, logically resolver should resolve, it can get really messy if at any point explicitly provided things would diverge (bug, whatever), so it's also safer.

Reply:
100%, will do

Reply:
fixed